README.md

```diff
@@ -3,45 +3,92 @@
+	$ redbiom search observations --context deblur-100nt --category empo_3 <space delimited list of observation IDs>
 
 That was fun. So now let's go a little further. Perhaps you are interested not just in where those sequences are found in, but also in the samples themselves for a meta-analysis. To pull out all the samples associated with your IDs of interest, and construct a BIOM table, you can do the following:
 
-	$ ./redbiom fetch observations --output some/path.biom <space delimited list of observation IDs>
+	$ redbiom fetch observations --context deblur-100nt --output some/path.biom <space delimited list of observation IDs>
+
+...but you probably also want the metadata! Once you have your table, you can obtain it by passing the table back in. This will attempt to grab the metadata (only the columns in common at the moment) for all samples present in your table. Note that we do not have to specify a context here as the sample metadata are context independent:
+
+	$ redbiom fetch sample-metadata --output some/path.txt --table some/path.biom 
+
+# Design decisions
 
-...but you probably also want the metadata! Once you have your table, you can obtain it by passing the table back in. This will attempt to grab the metadata (only the columns in common at the moment) for all samples present in your table:
+There are a few design decisions in place which deviate from some other typical Python projects. First off, the majority of `import`s are deferred. The motivating force here is to minimize overhead on load as to provide a responsive user interface -- deferred imports are the most straight forward way to accomplish that goal. 
 
-	$ ./redbiom fetch sample-metadata --output some/path.txt --table some/path.biom 
+The test harness is broken into multiple components, and are driven by a `Makefile`. This was done initially to be pragmatic as it was easier to write integration tests than unit tests for the `click` commands. That, and I didn't have wifi at the time in order to refer to the docs. These tests can be found in `test.sh` which is composed of "positive" tests and `test_failures.sh` which is composed of "negative" tests. The difference being that the positive tests will fail if any command results in a nonzero exit status, whereas the negative tests expect a nonzero exit status (and really, the decision was to avoid unsetting "-e"). Additional tests which validate some of the Redis contents can be found in `redbiom/tests/test_rest.py`. These are neither unit tests nor integration tests, but simply exercise the behind-the-scenes REST interface. Last, there are some unit tests placed under `redbiom/tests/test_util.py`. Tests have additionally not been written _yet_ for the `requests` calls also due to a lack of access to documentation at the time of writing, however due to the integration tests, there is a high confidence the methods are operating correctly sans some edge cases.
```

Comment:
Those tests are already part of this PR, I just didn't have the chance to revise the readme again.